aws/resource_aws_elasticsearch_domain_test.go

```diff
@@ -1592,6 +1680,100 @@ resource "aws_elasticsearch_domain" "test" {
 `, randInt)
 }
 
+func testAccESDomainConfig_AdvancedSecurityOptionsUserDb(randInt int) string {
+	var user = ""
+	var options = `
+advanced_security_options {
+	enabled = true
+	internal_user_database_enabled = true
+	master_user_options {
+		master_user_name = "testmasteruser"
+		master_user_password = "Barbarbarbar1!"
+	}
+}`
+
+	return fmt.Sprintf(`
+%s
+
+resource "aws_elasticsearch_domain" "test" {
+  domain_name           = "tf-test-%d"
+  elasticsearch_version = "7.1"
+
+  cluster_config {
+    instance_type = "r5.large.elasticsearch"
+  }
+
+	%s
+
+  encrypt_at_rest {
+    enabled = true
+  }
+
+  domain_endpoint_options {
+    enforce_https = true
+    tls_security_policy = "Policy-Min-TLS-1-2-2019-07"
+  }
+
+  node_to_node_encryption {
+    enabled = true
+  }
+
+  ebs_options {
+    ebs_enabled = true
+    volume_size = 10
+  }
+}
+`, user, randInt, options)
+}
+
+func testAccESDomainConfig_AdvancedSecurityOptionsIAM(randInt int) string {
+	var user = fmt.Sprintf(`
+resource "aws_iam_user" "es_master_user" {
+	name = "esmasteruser%d"
+}`, randInt)
+	var options = `
+advanced_security_options {
+	enabled = true
+	internal_user_database_enabled = false
+	master_user_options {
+		master_user_arn = "${aws_iam_user.es_master_user.arn}"
+	}
+}`
+
+	return fmt.Sprintf(`
+%s
+
+resource "aws_elasticsearch_domain" "test" {
+  domain_name           = "tf-test-%d"
```

Comment:
based on the above comment, pass here the test generated `rName` this can be done for the iam user name as well